File: src/main/java/bad/robot/radiate/teamcity/Projects.java
package bad.robot.radiate.teamcity;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Iterator;

class Projects extends TeamCityObject implements Iterable<Project> {

    @SerializedName("project")
    private Collection<Project> projects = new ArrayList<Project>();

    Projects(Collection<Project> projects) {
        this.projects = projects;
    }

    @Override
    public Iterator<Project> iterator() {
        return projects.iterator();
    }
}

File: src/main/scala/bad/robot/radiate/teamcity/ProjectS.scala
package bad.robot.radiate.teamcity

import bad.robot.radiate.HypermediaS

case class ProjectScala(id: String, name: String, href: String) extends TeamCityObjectS with HypermediaS with Iterable[BuildTypeScala] {

  def iterator = List[BuildTypeScala]().iterator

  override def toString() = s"$name ($id)"
}
